design/apidsl/media_type.go

```diff
@@ -376,9 +376,17 @@ func Link(name string, view ...string) {
 // CollectionOf creates a collection media type from its element media type. A collection media
 // type represents the content of responses that return a collection of resources such as "list"
 // actions. This function can be called from any place where a media type can be used.
+//
 // The resulting media type identifier is built from the element media type by appending the media
 // type parameter "type" with value "collection".
-func CollectionOf(v interface{}, apidsl ...func()) *design.MediaTypeDefinition {
+//
+//  CollectionOf(BottleMedia)  // If the identifier of BottleMedia is "vnd.goa.bottle',
+//                             // Content-Type will be "vnd.goa.bottle; type=collection".
```

Comment:
Should be `identifier` not `Content-Type` I think. By default the identifier is used to build the content type but that can be overridden with `ContentType`. So maybe something like:
```go
// Examples:
//
//   // Define a collection media type using the default generated identifier
//   // (e.g. "vnd.goa.bottle; type=collection" assuming the identifier of BottleMedia
//   // is "vnd.goa.bottle") and the default views (i.e. inherited from the BottleMedia
//   // views).
//   var col = CollectionOf(BottleMedia)
//
//   // Another collection media type using the same element media type but defining a
//   // a different default view.
//   var col2 = CollectionOf(BottleMedia, "vnd.goa.bottle.alternate; type=collection;", func() {
//        View("default", func() {
//           Attribute("id")
//           Attribute("name")
//        })
//   })
```